contracts/Root.sol

```diff
@@ -19,6 +19,13 @@ contract Root is Ownable {
     uint16 constant private TYPE_TXT = 16;
     uint16 constant private TYPE_SOA = 6;
 
+    bytes4 constant private INTERFACE_META_ID = bytes4(keccak256("supportsInterface(bytes4)"));
+    bytes4 constant private ROOT_REGISTRATION_ID = bytes4(
```

Comment:
why are we combining these?

Reply:
It's an EIP165 interface ID, which consists of the XOR of the function IDs of all the functions in the interface.